Flagsmith.Client.Test/FlagsmithTest.cs

```diff
@@ -391,6 +434,31 @@ public async Task TestFlagsmithUsesOfflineHandlerIfSetAndNoAPIResponse()
             Assert.Equal("offline-value", await identityFlags.GetFeatureValue("some_feature"));
         }
 
+        [Fact]
+        public async Task TestFlagsmithUsesTheAPIResponseEvenIfTheOfflineHandlerIsSet()
+        {
+            // Given
+            var environment = JObject
+                .Parse(File.ReadAllText("../../../data/offline-environment.json"))
+                .ToObject<EnvironmentModel>();
+
+            var mockOfflineHandler = new Mock<BaseOfflineHandler>();
+            mockOfflineHandler.Setup(h => h.GetEnvironment()).Returns(environment);
+            var mockHttpClient = HttpMocker.MockHttpResponse(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(Fixtures.ApiFlagResponse),
+            });
+
+            // When
+            FlagsmithClient flagsmithClientTest = new FlagsmithClient(Fixtures.ApiKey, httpClient: mockHttpClient.Object, offlineHandler: mockOfflineHandler.Object);
+
+            // Then
+            var environmentFlags = await flagsmithClientTest.GetEnvironmentFlags();
+            Assert.True(await environmentFlags.IsFeatureEnabled("some_feature"));
+            Assert.Equal("some-value", await environmentFlags.GetFeatureValue("some_feature"));
```

Comment:
Could we explicitly check that the `mockOfflineHandler` wasn't called here as well perhaps?

Reply:
Actually, it should only make a call based on this condition.
https://github.com/Flagsmith/flagsmith-dotnet-client/blob/0a2ba7950cee37f676d2b4db16f6f52884f83dae/Flagsmith.FlagsmithClient/FlagsmithClient.cs#L130

Reply:
Yes, but the test should verify that, right?

Reply:
I added 2 checks:

- Verify that the endpoint `api/v1/flags` is called once to retrieve its values.
- Check that the flag value is not the same as the value in the environment file.